### ISSUE TITLE
File collection media/public folder

### ISSUE THREAD

**Issue by BlueFrog130** [sveltia/sveltia-cms#155]
As of now, the media_folder and public_folder properties are not respected for file collections, is this something that is planned on being implemented?

I don't see any documentation on [Decap](https://decapcms.org/docs/collection-file/) specifically for media/public folders in file collections.

I haven't done much digging, but I assume the process for saving file collection assets would be the same as folder collections.

**Comment by kyoshino** [comment]
@BlueFrog130 Thanks for the heads-up! Many things are unclear in the Decap docs. I’m aware of at least one issue with file collections:

- https://github.com/decaporg/decap-cms/issues/978

Regardless of the Decap implementation, I’m happy to get rid of the limitation in Sveltia. 🙂

**Comment by kyoshino** [comment]
@BlueFrog130 Can you paste a snippet of your configuration to make sure I’m on the same page as you?

**Comment by BlueFrog130** [comment]
@kyoshino 

Here is a snippet that I am using:

```yml
collections:
  - label: Settings
    name: settings
    icon: Settings
    # Should the folder config be here
    media_folder: /src/uploads/settings 
    public_folder: /uploads/settings
    files:
      - label: Website Settings
        name: settings
        file: content/settings/settings.json
        # or here?
        media_folder: /src/uploads/settings
        public_folder: /uploads/settings
        editor:
          preview: false
        fields:
          ...
```

**Comment by kyoshino** [comment]
Thanks! The `media_folder` and `public_folder` options for the collection itself should work. These options for each file are not supported at this time.

**Comment by kyoshino** [comment]
```yaml
  - label: Settings
    name: settings
    icon: Settings
    media_folder: /src/uploads/settings 
    public_folder: /uploads/settings
    files:
      - label: Website Settings
        name: settings
        file: content/settings/settings.json
        editor:
          preview: false
        fields:
          - name: title
            label: Title
            widget: string
          - name: image
            label: Image
            widget: image
```

Here’s a screenshot of my local testing:

<img width="996" alt="image" src="https://github.com/sveltia/sveltia-cms/assets/2929505/fb4347e7-a73d-42eb-81de-124ce9e3b697">

**Comment by BlueFrog130** [comment]
Odd, my image uploads are going in the default media folder

```yml
media_folder: /src/uploads
public_folder: /uploads
collections:
  - label: Settings
    name: settings
    icon: Settings
    media_folder: /src/uploads/settings
    public_folder: /uploads/settings
    files:
      - label: Website Settings
        name: settings
        file: /content/settings/settings.json
        editor:
          preview: false
        fields:
          - label: Site Title
            name: title
            widget: string
          - label: Site Description
            name: description
            widget: text
          - label: Site Favicon
            name: favicon
            widget: image
```

![image](https://github.com/sveltia/sveltia-cms/assets/50666443/7f24a08c-0d8e-4503-b782-abceab636abf)

**Comment by kyoshino** [comment]
Perhaps due to the backslash weirdness on Windows. Will check.

**Comment by kyoshino** [comment]
I was able to reproduce the problem locally. For some reason, it fails when a file collection is the first collection. Fix coming.

**Comment by BlueFrog130** [comment]
@kyoshino

You the best. This project is awesome and I plan to use it a bunch.

Keep up the great work 👍

**Comment by kyoshino** [comment]
The fix landed in [v0.34.2](https://github.com/sveltia/sveltia-cms/releases/tag/v0.34.2)!
